Player iterator

Kanjirito opened the pull request:
PR for changes mentioned in #64.

The 2 main things the PR includes are:

1. `Player` getting it's lifetime removed
2. The `PlayerIter` being added (and all of the find methods switched to using it)

The lifetime has been removed by getting rid of the `BusName` and `Path` structs in `Player`. `BusName` has been replaced with a regular `String` that's getting converted whenever needed and `Path` has been removed completely. The reason why I decided to just remove `Player.path` is that it's a constant `str` anyway and since this is a MPRIS library there is no reason to use a custom path that doesn't follow MPRIS specs. Direct quote from the [MPRIS2 spec](https://specifications.freedesktop.org/mpris-spec/latest/#Entry-point):
> The media player **must** expose the **/org/mpris/MediaPlayer2** object path

Also, removing the lifetime is a breaking change anyways and I highly doubt anyone was calling `Player::new` directly so I don't believe this will be a big problem. There is the option to add `Player.path` back and just make it a `String` but I don't see the benefit of that.

The `PlayerIter` is a simple struct that implements `Iterator` (and some other iterator traits). Internally it just iterates over the return value of `PlayerFinder::all_player_buses` and returns a `Result<Player, DBusError>` for each of the bus names. I decided to use `std::vec::IntoIter` directly instead of using he `Vec<String>` simply because it's:
 1. Easier (just call `next()`)
 2. "Probably very efficient since it's in the standard library"

The other way I would have done it is get the `Vec<String>`, reverse it and then do `self.buses.pop()` on each iteration. That way you avoid popping from the front and still keep the alphabetical order. I'm not completely sure if this was the right decision so let me know if you think there's a better way.


---

There are also some other smaller changes int he PR:
 1. Very slight documentation changes
 2. The error messages have been fixed. The formatting in the error macro was wrong and instead of printing the actual error message it was printing literally just `0`
 3. Added a player timeout field on `PlayerFinder`. This is a value that will be passed to every `Player` created from that finder. It basically lets you to set a new default `Player.timeout_ms` for found players.

Kanjirito committed:
Remove `Player` lifetime

This commit removes the lifetime on `Player` by changing it's `bus_name`
and `path` field types.
The `bus_name` field is replaces with a `String` and the `path` field
is removed completely because it's a constant value anyway.

Kanjirito committed:
Implement PlayerIter

Adds an iterator for `PlayerFinder` that will iterate over the `Player`s
it finds.
Internally just uses a `vec::IntoIter` over the bus names and tries to
turn each one into a `Player`

Kanjirito committed:
Fix error messages

Currently all of the errors will print "0" instead of the actual error
message, this commit fixes that.

Kanjirito committed:
Add additional iterator traits to `PlayerIter`

Kanjirito committed:
Change finder methods to use `PlayerIter` internally

Mange commented:
I quite like all the proposed changes. It makes much more sense than my initial implementation!

I'm willing to merge this. First, would you mind adding entries to the Changelog? Preferably inside the actual commits that did the change so the history is clean.

Kanjirito committed:
Add timeout option for `PlayerFinder`

Adds a simple setting to change the DBus timeout that all the
`Player`s will inherit.

Kanjirito commented:
Took me a while to figure how to do it but I think everything is correct now.

Mange commented:
Excellent work! :heart:
